mparticle-integrations/mparticle-javascript-integration-adwords
# feat: Add support for Consent State Updates

## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
 - Adds a check to initForwarder and processEvents so that if the consent state changes, we alert Google via their `gtag` as per their [Consent Mode Documention](https://developers.google.com/tag-platform/security/guides/consent)

 ## Testing Plan
 - Set up Consent Mapping and Privacy Configuration via the mParticle UI
 - Using our [Data Privacy Controls] instructions, toggle some of the mapped consent states via the developer console
 - Verify that mapped events appear in `window.dataLayer` via browser developer console as updates with updated values

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-6133
